src/postprocess/chi0.jl

```diff
@@ -155,58 +151,79 @@ is false, only compute excitations inside the provided orbitals.
-            # δψn = P ψn + Q ψn
-            # we compute Pψn explicitly by sum over states, and
-            # Q δψn by solving the Sternheimer equation
-            # (H-εn) Q δψn = -Q δV ψn
-            # where Q = sum_n |ψn><ψn|
-
-            # explicit contributions
-            for m = 1:size(ψ[ik], 2)
-                εmk = eigenvalues[ik][m]
-                ddiff = Smearing.occupation_divided_difference
-                ratio = filled_occ * ddiff(model.smearing, εmk, εnk, εF, temperature)
-                (n != m) && (abs(ratio) < droptol) && continue
-                ψmk = @view ψ[ik][:, m]
-                ψmk_real = G_to_r(basis, basis.kpoints[ik], ψmk)
-                # ∑_{n,m != n} (fn-fm)/(εn-εm) ρnm <ρmn|δV>
-                ρnm = conj(ψnk_real) .* ψmk_real
-                weight = dVol * dot(ρnm, δV)
-                δρk .+= real(ratio .* weight .* ρnm)
-            end
-
-            if sternheimer_contribution
-                # Compute the contributions from uncalculated bands
-                fnk = filled_occ * Smearing.occupation(model.smearing, (εnk-εF) / temperature)
-                abs(fnk) < eps(T) && continue
-                rhs = r_to_G(basis, basis.kpoints[ik], .- δV .* ψnk_real)
-                norm(rhs) < 100eps(T) && continue
-                δψnk = sternheimer_solver(ham.blocks[ik], ψ[ik], ψnk, εnk, rhs;
-                                          kwargs_sternheimer...)
-                δψnk_real = G_to_r(basis, basis.kpoints[ik], δψnk)
-                δρk .+= 2 .* fnk .* real(conj(ψnk_real) .* δψnk_real)
-            end
+        hamk = ham.blocks[ik]
+        εk = eigenvalues[ik]
+        ψk = ψ[ik]
+
+        # One δρk accumulator per thread
+        δρk = [zero(δV) for tid = 1:Threads.nthreads()]
+        Threads.@threads for n = 1:size(ψ[ik], 2)
+            tid = Threads.threadid()
+            add_response_from_band_!(δρk[tid], hamk, εk, ψk, n, εF, δV,
+                                     temperature, droptol, sternheimer_contribution,
+                                     kwargs_sternheimer)
         end
-        accumulate_over_symops!(δρ_fourier, r_to_G(basis, complex(δρk)), basis, basis.ksymops[ik])
+        accumulate_over_symops!(δρ_fourier, r_to_G(basis, complex(sum(δρk))),
+                                basis, basis.ksymops[ik])
     end
     δρ = real(G_to_r(basis, δρ_fourier))
     count = sum(length(basis.ksymops[ik]) for ik in 1:length(basis.kpoints))
     δρ ./= count
 
     # Add variation wrt εF
     if temperature > 0
+        dVol = basis.model.unit_cell_volume / prod(basis.fft_size)
         ldos = LDOS(εF, basis, eigenvalues, ψ, temperature=temperature)
-        dos = DOS(εF, basis, eigenvalues, temperature=temperature)
+        dos  = DOS(εF, basis, eigenvalues, temperature=temperature)
         δρ .+= ldos .* dot(ldos, δV) .* dVol ./ dos
     end
 
     from_real(basis, δρ .* normδV)
 end
+
+
+function add_response_from_band_!(δρk, hamk, εk, ψk, n, εF, δV,
```

Comment:
OK the underscore madness has to stop. Do we use underscores before the function name, after the function name, or not at all?

Reply:
Add a very short doc as to what this function is doing. Maybe also put the "n" argument a bit before so that it's clearer it's a crucial input